src/core/action/edit/space_domain/ApplyOtsuThresholdEstimationAction.java

```diff
@@ -0,0 +1,132 @@
+package core.action.edit.space_domain;
+
+import core.service.MatrixService;
+import core.service.generation.HistogramService;
+import domain.Histogram;
+import domain.automaticthreshold.OtsuThresholdImage;
+import domain.customimage.CustomImage;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class ApplyOtsuThresholdEstimationAction {
+
+    private HistogramService histogramService;
+    private MatrixService matrixService;
+
+    public ApplyOtsuThresholdEstimationAction(HistogramService histogramService, MatrixService matrixService){
+        this.histogramService = histogramService;
+        this.matrixService = matrixService;
+    }
+
+    public OtsuThresholdImage execute(CustomImage customImage){
+        //calculo histograma
+        Histogram histogram = this.histogramService.create(customImage);
+
+        //calculo las sumas acumulativas del histograma
+        Double[] cumulativeProbabilities = new Double[256];
+        for (int i = 0; i <= 255; i++){
+            cumulativeProbabilities[i] = this.cumulativeProbability(histogram, i);
+        }
+
+        //para cada t desde 0 a 255 tengo dos grupos C1 (menor al umbral) y C2 (mayor al umbral)
+
+        //calculo las medias acumulativas (similar a las sumas acumulativas) pero multiplico a pi(p sub i) * i, yendo i de 0 a t
+        Double[] cumulativeMeans = new Double[256];
+        for (int t = 0; t <= 255; t++){
+            cumulativeMeans[t] = this.cumulativeMean(histogram, t);
+        }
+
+        //calculo la media global igual que la acumulativa, pero t sería 255
+        Double globalCumulativeMean = cumulativeMeans[255];
+
+        //calculo varianza entre clases para cada t: (mg * P1(t) - m(t))² / P1(t) * (1 - P1(t))
+        Double[] variances = new Double[256];
+        for (int t = 0; t <= 255; t++){
+            Double tCumulativeProbability = cumulativeProbabilities[t];
+            Double tCumulativeMean = cumulativeMeans[t];
+            variances[t] = Math.pow(((globalCumulativeMean * tCumulativeProbability) - tCumulativeMean), 2)
+                            / (tCumulativeProbability * (1 - tCumulativeProbability));
+        }
+
+        //busco con que t me dio una varianza mas grande. Pueden ser mas de uno
+        List<Integer> tMaxValues = this.searchTThatGenerateMaximumVariance(variances);
```

Comment:
agregaste una "T" de mas en el nombre del metodo?

Reply:
No, dice "search T (es el valor de umbral) that generate maximum variance"